### PR TITLE
Sidebar Adding Icon Functionality

### PR DESCRIPTION
**Pull request by wais-hundekar**
## Notion ticket link

[Update Sidebar Component to Render Different Icon](https://www.notion.so/uwblueprintexecs/47bf4727a2444bccad4a91952898b8f3?v=9658436bf9be409e90581631a7155f16&p=90dd4f46f0954c1d818d6fd4f131d3e8&pm=s)

## Implementation description
* Making Icon a prop so that icon can be changed and added dynamically

## Steps to test
1. Ensure functionality works on various icons and formatted properly

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/types/PageTypes.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
+import React from "react";
+
 interface Page {
   title: string;
   url: string;
+  icon?: () => React.ReactElement;
```
Comment (chenjanet) [inline frontend/src/types/PageTypes.ts:6]: Is there a particular reason why this is an optional field?

Comment (joyce-shi) [reply frontend/src/types/PageTypes.ts:6]: +1! let's make this a mandatory field & maybe we can add a default icon in the `Sidebar` component!

**Review by joyce-shi — APPROVED** [review]
+1 to Janet's comment, but other than that LGTM! Feel free to merge after that! 🚀

File: frontend/src/types/PageTypes.ts (hunk 1 above)
Comment (wais-hundekar) [reply frontend/src/types/PageTypes.ts:6]: Oh I have it added like that because certain pages become subpages, which do not have icons

Comment (wais-hundekar) [reply frontend/src/types/PageTypes.ts:6]: It's like why the subpages field is also optional

Comment (joyce-shi) [reply frontend/src/types/PageTypes.ts:6]: oh jk you right! forgot about that

**Pull request merged**